new health facility and other infrastructure terms

ddooley opened the pull request:
Re. Issue #1070 
bridges, transport terms, utensils, food processing and agriculture infrastructure terms, some materials.
A redo of pull request 1071.

ddooley commented:
I believe the remaining warnings and errors are unrelated to this pull request.

pbuttigieg commented:
>I believe the remaining warnings and errors are unrelated to this pull request.

Yes, these should be resolved as you just merged master into your branch - @turbomam figured out that the order of the arguments in the DOSDP command (the "generate" arg) was breaking things sometimes, randomly.

turbomam commented:
Shoot, I had intended to push a fix for the Makefile regarding the DOSDP arg order. @pbuttigieg good thing we didn't delete my merged branch yet. I'll see if I can dig that Makefile edit back up.

Basically, the DOSDP subcommand (like generate) should come before any named options.

turbomam commented:
Here's the [DOSDP invocation](https://github.com/EnvironmentOntology/envo/blob/Jagadish-NMDC-terms-no-makefile/src/envo/Makefile#L542-L545) that worked in my modified Makefile

ddooley commented:
I have updated the envo-edit.owl file, and removed a few stray top-level terms.  There is an additional term import added to PATO and OBI import files; PATO one will take care of that top level 0040009 term. The other FoodOn top level numbered terms should disappear if ENVO were to do a new FoodOn import this weekend, and as well I've removed a few overbearing FoodOn equivalencies which may eliminate occurrence of some other top level FoodOn terms. 

![image](https://user-images.githubusercontent.com/4000582/118896829-fb565180-b8bd-11eb-8c22-8c8a9b2c68a5.png)

pbuttigieg commented:
Thanks @ddooley - we should do the import in this PR. If you plan a FOODON release this weekend, it makes sense to wait

ddooley commented:
I will try to finish FOODON release tomorrow!

ddooley commented:
The FoodOn v0.5 release has been done, which may help with some top-level stray ENVO terms on build.

pbuttigieg commented:
@ddooley - I don't see the FOODON import in this PR refreshed to match the new release. We'd need that to close this. Let me know if you'd do that or if you need backup.

ddooley commented:
@pbuttigieg  I'm a little confused about what you said.  Wasn't the plan to do a FoodOn release referesh separately from this pull request?  If you meant to do a FoodOn release refresh within this pull request by running a build or a refreshed import on FoodOn import file via ontofox or something, let me know because yes I will need help to see how that is done in ENVO context.

pbuttigieg commented:
>@pbuttigieg I'm a little confused about what you said. Wasn't the plan to do a FoodOn release referesh separately from this pull request? If you meant to do a FoodOn release refresh within this pull request by running a build or a refreshed import on FoodOn import file via ontofox or something, let me know because yes I will need help to see how that is done in ENVO context.

Hi @ddooley

You're correct - the FOODON release is separate and on FOODON's side. Now that that's done, we'll re-run the `make foodon_import.owl`  target in our makefile to refresh the import for this PR. 

I'll note the code here, but it will generally follow this procedure: https://github.com/EnvironmentOntology/envo/wiki/Import-terms-from-other-ontologies

pbuttigieg commented:
The build failing is linked to a "too many requests" error [here](https://travis-ci.org/github/EnvironmentOntology/envo/builds/774512131#L163-L194)

@cmungall is there something up that we should be concerned about generally?

ddooley commented:
Any update on this?  Is the June 14th issue resolved?

cmungall commented:
copied from other ticket:

@ddooley 
> Shall I make the label and parent changes on current pull request (which should trigger a new approval process?) > OR I'll make the changes asap on a new request after pull goes in?

@kaiiam:
> I feel like it'd be better not to do the latter, but that's just me. I'd suggest either fixing the existing PR or regenerating a new one if that's easier. Whatever's easiest for you. Were you perchance using the > https://github.com/EnvironmentOntology/envo/wiki/ENVO-Robot-template-and-merge-workflow for a robot template of terms? Might help.

I agree with Kai, I think it's easier to tweak this PR but I am also OK with merging this then making a quick patch up merge, whatever works.

ddooley commented on src/envo/envo-edit.owl:
```diff
@@ -39305,6 +39601,2561 @@ AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <ht
+# Class: <http://purl.obolibrary.org/obo/ENVO_03501302> (cork (FOODON:03500046))
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <https://en.wikipedia.org/wiki/Cork_(material)>) <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/ENVO_03501302> "Plant matter consisting of an impermeable buoyant material, the phellem layer of bark tissue."@en)
+AnnotationAssertion(<http://www.geneontol
```
All (FOODON: ...) were accidental.  Foodon id was meant just to be a reference to term to replace in foodon.

ddooley commented on src/envo/envo-edit.owl:
```diff
@@ -39305,6 +39601,2561 @@ AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <ht
+# Class: <http://purl.obolibrary.org/obo/ENVO_03501302> (cork (FOODON:03500046))
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <https://en.wikipedia.org/wiki/Cork_(material)>) <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/ENVO_03501302> "Plant matter consisting of an impermeable buoyant material, the phellem layer of bark tissue."@en)
+AnnotationAssertion(<http://www.geneontol
```
fixed

ddooley commented on src/envo/envo-edit.owl:
```diff
@@ -39305,6 +39601,2561 @@ AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <ht
+AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/ENVO_03501218> "toilet bowl"@en)
+SubClassOf(<http://purl.obolibrary.org/obo/ENVO_03501218> <http://purl.obolibrary.org/obo/ENVO_01000989>)
+SubClassOf(<http://purl.obolibrary.org/obo/ENVO_03501218> ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/BFO_0000050> <http://purl.obolibrary.org/obo/ENVO_01000516>))
+SubClassOf(<http://purl.obolibrary.org/obo/ENVO_03501218> ObjectSomeValuesFrom(<http://purl.obolibrary.org/obo/RO_0000086> <http://purl.obolibrary.org/obo/PATO_0040009>))
```
removed pato quality

kaiiam commented:
Quite the PR! Looking better in regard to the above comments about anti-patterns and labels. Definitions seem to follow ENVO conventions (I admit I don't have time to look though all of this in the depth it deserves). I agree with @cmungall's thoughts:

> I am not totally sure ENVO is the best long term home for some of these super-specific concepts. However, as of now I do not have a concrete recommendation.

I think this could be the start of a whole new ontology of manufactured products and anthropogenic (human built) environments. Would that be better? Or should ENVO include terms like `spoon` and `recreational cruise ship` and `medical clinic` etc. I can see arguments both ways. Less ontologies means less re-importing of terms and technical issues, but having a separate ontology for something like manufactured products and human constructions would better delineate scope disciplines covered e.g., leaving earth and environmental sciences within ENVO. I understand anthropogenic environments have always been in ENVO's purview perhaps it would be manufactured products that should be housed elsewhere than ENVO? 

Rought proposal idea: Manufactured Product Ontology (MPO) could be the home for the current ENVO:00003074 `manufactured product` hierarchy and be responsible for all variations of spoons and squeegees and other endless crap (pardon my language) that humans produce. Meanwhile ENVO could remain the host for all environments natural and anthropogenic and be the home for terms like `processing plant` and `hospital unit facility` etc. Just a suggestion the alternative being that ENVO absorbs all of this. 

Thoughts?

kaiiam commented:
It's just an idea although @ddooley @matentzn and I have bounced it around rhetorically. Just something to consider I'm obviously in favor and willing to help set it up, but only if people think it's a useful thing for OBO overall.

pbuttigieg commented on src/envo/envo-edit.owl:
```diff
@@ -39305,6 +39601,2561 @@ AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <ht
+# Class: <http://purl.obolibrary.org/obo/ENVO_03501302> (cork (FOODON:03500046))
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <https://en.wikipedia.org/wiki/Cork_(material)>) <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/ENVO_03501302> "Plant matter consisting of an impermeable buoyant material, the phellem layer of bark tissue."@en)
+AnnotationAssertion(<http://www.geneontol
```
https://github.com/EnvironmentOntology/envo/pull/1105/files#diff-9a5b50d513c6b7f58aeb2d37743df7b38d8552400cb9daed1ff48a7f0b7e2b37R41671-R41677

pbuttigieg commented:
I'm fine with merging in this in and tweaking later. There are multiple issues to discuss here and also some policy questions to consider.

kaiiam commented:
Sounds good. If in the future it makes sense to scope the manufacture product hierarchy elsewhere we could deal with that then. For now ENVO will house those terms.

pbuttigieg commented:
>Sounds good. If in the future it makes sense to scope the manufacture product hierarchy elsewhere we could deal with that then. For now ENVO will house those terms. 

Yes, and I wouldn't move them until there's a viable ontology up and running

cmungall commented on src/envo/envo-edit.owl:
```diff
@@ -39305,6 +39601,2552 @@ AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <ht
+
+# Class: <http://purl.obolibrary.org/obo/ENVO_03501102> (polytunnel)
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <http://purl.obolibrary.org/obo/ENVO_>) <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/ENVO_03501102> "A tunnel made from steel and covered in polyethylene used for plant cultivation."@en)
```
How did this happen?

cmungall commented on src/envo/envo-edit.owl:
```diff
@@ -39305,6 +39601,2552 @@ AnnotationAssertion(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <ht
+
+# Class: <http://purl.obolibrary.org/obo/ENVO_03501109> (subway train)
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <http://purl.obolibrary.org/obo/ENVO_>) <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/ENVO_03501109> "A passenger train which runs along an undergroud rail system."@en)
```
multiple cases of this

matentzn replied:
@ddooley I think it is probably best to use oboInOwl:source (or dc:source), whatever ENVO uses, instead of hasDBXref..

ddooley replied:
If I'm not mistaken ENVO uses hasdbxref as annotation on definitions to point to definition source.  (Personally, given that a term should have only one definition I have been fine with the IAO:0000118 definition source annotation on the term itself that other projects use).  At any rate I did a pull request https://github.com/EnvironmentOntology/envo/pull/1309  to get rid of the bad hasDBXrefs.

kaiiam replied:
>  ENVO uses hasdbxref as annotation on definitions to point to definition source. 

Yes that's our standard, theres always more cleanup but thanks @ddooley for cleaning up in https://github.com/EnvironmentOntology/envo/pull/1309 much obliged!

cmungall reviewed (commented):
there are multiple syntactic issues. Not sure why this did not get caught

kaiiam commented:
Yes we should probably fix all this before the next release.
